src/dbseeder/programs.py

```diff
@@ -134,116 +144,192 @@ class WqpProgram(object):
-        parent_folder = join(file_location, self.datasource)
+        self.db = db
 
-        if not isdir(parent_folder):
-            raise Exception('Pass in a location to the parent folder that contains WQP. {}'.format(parent_folder))
+        #: if file_location is None then we are updating
+        if file_location is not None:
+            #: check that file_location exists wqp/results and wqp/stations
+            parent_folder = join(file_location, self.datasource)
 
-        results_folder = join(parent_folder, 'Results')
-        stations_folder = join(parent_folder, 'Stations')
+            if not isdir(parent_folder):
+                raise Exception('Pass in a location to the parent folder that contains WQP. {}'.format(parent_folder))
 
-        if not isdir(results_folder or stations_folder):
-            raise Exception('WQP folder is missing Results or Stations child folders containing csv files.')
+            results_folder = join(parent_folder, 'Results')
+            stations_folder = join(parent_folder, 'Stations')
 
-        #: all is well set the folders for seeding later on
-        self.results_folder = results_folder
-        self.stations_folder = stations_folder
+            if not isdir(results_folder or stations_folder):
+                raise Exception('WQP folder is missing Results or Stations child folders containing csv files.')
 
-        self.db = db
+            #: all is well set the folders for seeding later on
+            self.results_folder = results_folder
+            self.stations_folder = stations_folder
 
     def seed(self):
+        if not hasattr(self, 'results_folder') or not hasattr(self, 'stations_folder'):
+            raise Exception('You must pass a file location if you are seeding. Did you want to update?')
+
+        try:
+            self._seed_by_file()
+        finally:
+            if hasattr(self, 'cursor'):
+                del self.cursor
+
+    def update(self):
+        try:
+            last_updated = self._get_most_recent_result_date()
+
+            if not last_updated:
+                raise Exception('No last updated date')
+
+            #: get new results from wqp service
+            result_url = self._format_url(self.wqp_url, 'Result', last_updated)
+            #: group them as if they were read from querycsv
+            new_results = self._group_rows_by_id(HttpClient.get_csv(result_url))
+
+            #: find the station ids from the new results that aren't in the database
+            new_station_ids = self._find_new_station_ids(new_results)
+
+            if new_station_ids and len(new_station_ids) > 0:
+                print('of the new stations found, inserting {} unique'.format(len(new_station_ids)))
+
+                station_url = self._format_url(self.wqp_url, 'Station', last_updated)
+                new_stations = HttpClient.get_csv(station_url)
+
+                if not new_stations:
+                    return
```

Comment:
Should this through an error? There are new results with new station ids that don't have corresponding ids in the stations table from the wqp service. Or maybe there was something wrong with the stations service?